### DISCUSSION TITLE
Does the backend support PostgreSQL?

### DISCUSSION THREAD

**Discussion by deepminimal** [keephq/keep#599] [Q&A] [answered]
Hello!
I set postgresql env postgresql://keep:keep@POSTGRESQL_DB:5432/keep and get error
```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InvalidForeignKey) there is no unique constraint matching given keys for referenced table "alert"

[SQL: 
CREATE TABLE alertenrichment (
	enrichments JSON, 
	id UUID NOT NULL, 
	tenant_id VARCHAR NOT NULL, 
	timestamp TIMESTAMP WITHOUT TIME ZONE NOT NULL, 
	alert_fingerprint VARCHAR NOT NULL, 
	PRIMARY KEY (id), 
	FOREIGN KEY(tenant_id) REFERENCES tenant (id), 
	FOREIGN KEY(alert_fingerprint) REFERENCES alert (fingerprint)
) /*db_driver='psycopg2',db_framework='sqlalchemy%%3A0.41b0',traceparent='00-d7f2af33515860a1a80d3e7ea6ef2a8c-d3eb1dedf9e2117f-01'*/]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

**Comment by shahargl** [comment, marked as answer]
Hey! So I guess it doesn’t 😅 why do you need Postgres? Can you describe the use case?

**Reply by shahargl** [reply]
Like, I guess it would be easy to support that but wondering why MySQL isn’t sufficient

**Reply by deepminimal** [reply]
Hello! Thanks for you answer. 
We have a lot of PostgreSQL clusters based on Spilo (Patroni). In our products we using only PostgreSQL.

**Reply by shahargl** [reply]
Got ya. You can open a ticket and we will get to it probably soon. Also, it will be awesome if you’ll join our Slack so we can learn about your use case (https://slack.keephq.dev)

**Comment by talboren** [comment]
Hey @deepminimal, jumping in for a quick question to see if I can help you get this one sorted out fast: can you share the exact way you tried to deploy Keep? was it the via docker-compose and you just changed the connection string or was it something else? I'll try to see if we have a quick win for that to make it work for you.

**Reply by deepminimal** [reply]
With MySQL backend works fine. 
I use helm chart, in values https://github.com/keephq/keep/blob/4c7aaee247655a7d2e757576c46d37a84174c119/chart/keep/values.yaml#L19 changed to postgresql://keep:keep@POSTGRESQL_DB:5432/keep

but I have some problems, when I go to settings in ui got redirect loop, don’t understand how to setting socket, in your front wss set localhost need rebuild front with ingress wss url.  Soon I create issue of all problem I have.

**Reply by deepminimal** [reply]
Forgot about kubernetes, we don’t have any pvc in clusters.

**Reply by talboren** [reply]
> but I have some problems, when I go to settings in ui got redirect loop, don’t understand how to setting socket, in your front wss set localhost need rebuild front with ingress wss url. Soon I create issue of all problem I have.

Hmm.. sounds weird, have you followed the README.md and did: `kubectl port-forward svc/keep-websocket 6001:6001 &`? Happy to hop on a call and try to solve all your problems.

**Reply by talboren** [reply]
> Forgot about kubernetes, we don’t have any pvc in clusters.

PVC is mainly for the DB. I'll make postgresql work for you if this solves persistency for you.

**Reply by talboren** [reply]
FYI, you can also use `PUSHER_DISABLED=true` for both backend/frontend

**Comment by talboren** [comment]
@deepminimal [this PR](https://github.com/keephq/keep/pull/601) will resolve your issue with PostgreSQL.
